### DISCUSSION TITLE
Automatically create DB indexes once synced

### DISCUSSION THREAD

**Discussion by nrutledge** [CardanoSolutions/kupo#124] [Archives]
### What is your idea? Provide a use case.

Automatically create the DB indexes once kupo sync has caught up to the tip (or close to it). This would be applicable for any use case.

### Why is it a good idea?

This would eliminate the need to restart kupo with updated config in order to create the indexes.

### What is the current alternative and why is it not good enough?

Current alternative is to start kupo with `--defer-db-indexes` enabled initially and then restart with the option disabled once the sync has caught up. The problem with this approach is that it's far too easy to forget the second step or not even realize it is required.

For production deployments, this also introduces an unnecessary maintenance burden (either extra time working to automate these steps or manually monitoring the sync status, issuing PRs to change the config, etc.).

That being said, a highlight/bold warning about the need to disable the `--defer-db-indexes` in the docs would go a long way.

**Comment by KtorZ** [comment]
Actually done since v2.8.0
